app/celery/tasks.py

```diff
@@ -0,0 +1,57 @@
+import base64
+
+from botocore.exceptions import ClientError as BotoClientError
+from flask import current_app
+from notifications_utils.s3 import s3download, s3upload
+from notifications_utils.statsd_decorators import statsd
+
+from app import notify_celery, TaskNames, QueueNames
+from app.precompiled import sanitise_file_contents
+
+
+@notify_celery.task(name='sanitise-and-upload-letter')
+@statsd(namespace='template-preview')
+def sanitise_and_upload_letter(notification_id, filename):
+    current_app.logger.info('Sanitising notification with id {}'.format(notification_id))
+
+    try:
+        pdf_content = s3download(current_app.config['LETTERS_SCAN_BUCKET_NAME'], filename).read()
+        json_response = sanitise_file_contents(pdf_content)
+
+        # Only files that have failed validation have a message in the json_response
+        if json_response.get('message'):
+            validation_status = 'failed'
+        else:
+            validation_status = 'passed'
+            file_data = base64.b64decode(json_response['file'].encode())
+
+            # If the file already exists in S3, it will be overwritten
+            s3upload(
+                filedata=file_data,
+                region=current_app.config['AWS_REGION'],
+                bucket_name=current_app.config['S3_LETTER_SANITISE_BUCKET'],
```

Comment:
This sounds like a bucket of things to be sanitised rather than something that has been sanitised

Reply:
👍Updated to `SANITISED_LETTER_BUCKET_NAME`